onnxruntime/test/providers/cpu/ml/treeregressor_test.cc

```diff
@@ -844,5 +844,46 @@ TEST(MLOpTest, TreeRegressorTrueNodeBeforeNode) {
   test.Run();
 }
 
+TEST(MLOpTest, TreeRegressorBranchEq) {
```

Comment:
It is great to have this test here. Should we also add a test that checks that the folding of the chain of `BRANCH_EQ` nodes actually happened?